modules/fetch-article.js

```diff
@@ -1,9 +1,14 @@
 const db = require(`../db/connection`);
 
 module.exports = function fetchArticle(article_id) {
-    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    return db.query(
+        `SELECT articles.*, comments.article_id, count(*)::int AS comment_count
+        FROM comments 
+        LEFT JOIN articles ON comments.article_id = articles.article_id
+        WHERE comments.article_id = $1
+        GROUP BY comments.article_id, articles.article_id;`, [article_id])
         .then(({ rows: article }) => {
-            if (article.length === 0) {return Promise.reject({ code: 404, msg: 'Article not found' })}
+            if (article.length === 0) { return Promise.reject({ code: 404 }) }
             else { return article[0] }
-        });
+        }).catch((err) => {return Promise.reject({ code: err.code, msg: err.msg })});
```

Comment:
we shouldn't have catch blocks in the models, they should be kept for the controllers only and then the error can be passed via `next` to your error handling functions